x-pack/plugins/fleet/public/components/agent_enrollment_flyout/instructions.tsx

```diff
@@ -42,11 +40,16 @@ export const Instructions = (props: InstructionProps) => {
     refreshAgentPolicies,
   } = props;
   const fleetStatus = useFleetStatus();
-  const { isUnhealthy: isFleetServerUnhealthy } = useFleetServerUnhealthy();
+  const REFRESH_INTERVAL = 10 * 1000;
```

Comment:
I don't know if polling is the right solution here. Is there a way we can add `fleetStatus.refresh` to wherever we currently wait on the the existence of a Fleet Server? Because `fleetStatus` is a context object, we should be able to call refresh and update any other consumers of that context object from anywhere in the component tree.

Reply:
I tried using `fleetStatus.refresh` without the setInterval as a first try and didn't work as expected.
An issue that I had a lot with this component is that react signals too many rerenders, which is probably a sign that we are doing something weird. 

The solutions I could think of are basically 1- the polling that I implemented here 2- doing as many checks as possible in the parent components of the flyout, but since is called several times in the code it would be a lot of refactoring for a small bug fix.

Reply:
@criamico  if you put the `fleetStatus.refresh` in a `useEffect` with an empty array it should only refresh once no?
something like this 
`useEffect(() => fleetStatus.refresh(), []);